common/middleware/error.go

```diff
@@ -31,21 +33,39 @@ func LogError(id string, error_message interface{}) {
 func ErrorMiddleware(next http.Handler) http.Handler {
 	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
 		defer func() {
-			if rec := recover(); rec != nil {
+			if panic_object := recover(); panic_object != nil {
 				w.Header().Set("Content-Type", "application/json")
-				switch panic_error := rec.(type) {
-				case errors.APIError:
-					LogError(r.Header.Get("HackIllinois-Identity"), panic_error)
-					w.WriteHeader(panic_error.Status)
-					json.NewEncoder(w).Encode(panic_error)
+				switch panic_error := panic_object.(type) {
+				case errors.ApiError:
+					handleApiError(panic_error, w, r)
 				default:
-					LogError(r.Header.Get("HackIllinois-Identity"), rec)
-					w.WriteHeader(http.StatusInternalServerError)
-					json.NewEncoder(w).Encode(errors.APIError{Status: 500, Message: "Unknown Error"})
+					handleUnknownError(panic_error, w, r)
 				}
 			}
 		}()
-
 		next.ServeHTTP(w, r)
 	})
 }
+
+func handleApiError(err errors.ApiError, w http.ResponseWriter, r *http.Request) {
+	LogError(r.Header.Get("HackIllinois-Identity"), err)
+
+	w.WriteHeader(err.Status)
+
+	mode, is_mode_set := os.LookupEnv("DEBUG_MODE")
+
+	// Strip the raw error string if we're not in debug mode
+	if !is_mode_set || (is_mode_set && strings.ToLower(mode) != "true") {
+		err.RawError = ""
+	}
+
+	json.NewEncoder(w).Encode(err)
+}
+
+func handleUnknownError(err interface{}, w http.ResponseWriter, r *http.Request) {
+	LogError(r.Header.Get("HackIllinois-Identity"), err)
+
+	w.WriteHeader(http.StatusInternalServerError)
+
+	json.NewEncoder(w).Encode(errors.ApiError{Status: http.StatusInternalServerError, Type: "UNKNOWN_ERROR", Message: err.(string), RawError: ""})
```

Comment:
Add an `UnkownError` to `common/errors` so you don't need to directly construct an `ApiError`.

Also `err.(string)` will panic if `err` is not a `string`. The safest way to convert it to a string is `fmt.Sprintf("%v", err)`. Also I'd put that in `RawError` not `Message` and do the same thing with handling `config.DEBUG_MODE`.